src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -63,7 +58,21 @@ public HttpConnectionPoolManager(HttpConnectionSettings settings)
             _settings = settings;
             _maxConnectionsPerServer = settings._maxConnectionsPerServer;
             _pools = new ConcurrentDictionary<HttpConnectionKey, HttpConnectionPool>();
+
             // Start out with the timer not running, since we have no pools.
+            // When it does run, run it with a frequency based on the idle timeout.
+            if (settings._pooledConnectionIdleTimeout == Timeout.InfiniteTimeSpan)
```

Comment:
Not for this PR but it will be good to have a way control these (and other) knobs related to pooling.

Reply:
PooledConnectionIdleTimeout is exposed.  The thing missing is a knob for how often the scavenger runs.  I go back and forth on whether we actually want to expose a knob for such an implementation detail.  Definitely not for 2.1, though.